### PR TITLE
Update Azure.ResourceManager.RecoveryServicesSiteRecovery SDK to 2023-02-01

### PR DESCRIPTION
**Pull request by shesamian**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

This PR was generated from [stable 2023-02-01](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/service.json) of recoveryservicessiterecovery

[Microsoft.Azure.Management.RecoveryServices.SiteRecovery](https://www.nuget.org/packages/Microsoft.Azure.Management.RecoveryServices.SiteRecovery/) has been deprecated according to the readme so I didn't update it. I only updated [Azure.ResourceManager.RecoveryServicesSiteRecovery](https://www.nuget.org/packages/Azure.ResourceManager.RecoveryServicesSiteRecovery/).

### CODE REVIEW

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/Generated/Models/SecurityType.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.RecoveryServicesSiteRecovery.Models
+{
+    /// <summary> The target VM security type. </summary>
+    public readonly partial struct SecurityType : IEquatable<SecurityType>
```
Comment (archerzz) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/Generated/Models/SecurityType.cs:14]: The name is too generic. Consider appending RP prefix or renaming it (like `VirtualMachineSecurityType`).
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/Generated/Models/VMwareCbtEnableMigrationInput.cs [hunk 2]
```diff
@@ -79,6 +79,10 @@ public VMwareCbtEnableMigrationInput(string vmwareMachineId, IEnumerable<VMwareC
         public string TargetAvailabilityZone { get; set; }
         /// <summary> The target proximity placement group ARM Id. </summary>
         public string TargetProximityPlacementGroupId { get; set; }
+        /// <summary> The confidential VM key vault Id for ADE installation. </summary>
+        public string ConfidentialVmKeyVaultId { get; set; }
```
Comment (archerzz) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/Generated/Models/VMwareCbtEnableMigrationInput.cs:83]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/Generated/Models/VMwareCbtMigrationDetails.cs [hunk 3]
```diff
@@ -141,6 +152,10 @@ internal VMwareCbtMigrationDetails(string instanceType, string vmwareMachineId,
         public string TargetAvailabilityZone { get; }
         /// <summary> The target proximity placement group Id. </summary>
         public string TargetProximityPlacementGroupId { get; }
+        /// <summary> The confidential VM key vault Id for ADE installation. </summary>
+        public string ConfidentialVmKeyVaultId { get; }
```
Comment (archerzz) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/Generated/Models/VMwareCbtMigrationDetails.cs:156]: same. what's the format of this id?

**Review by archerzz — CHANGES_REQUESTED** [review]
You forgot to update the API signature. Run `eng\scripts\Export-API.ps1 {your_sdk}` to update it.

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/Generated/Models/VMwareCbtMigrationDetails.cs (hunk 3 above)
Comment (shesamian) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/Generated/Models/VMwareCbtMigrationDetails.cs]: `ConfidentialVmKeyVaultId` is now `ResourceIdentifier`.

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/Generated/Models/VMwareCbtEnableMigrationInput.cs (hunk 2 above)
Comment (shesamian) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/src/Generated/Models/VMwareCbtEnableMigrationInput.cs]: `ConfidentialVmKeyVaultId` is now `ResourceIdentifier`.

**Review by archerzz — APPROVED** [review]

**Pull request merged**